lib/options.nix

```diff
@@ -127,7 +127,17 @@ rec {
 
 
   /* Helper functions. */
-  showOption = concatStringsSep ".";
+  showOption =
+  let
+    # adds qoutes when string isn't valid nix identifier
+    # https://github.com/NixOS/nix/blob/a5e761dddb6b090b233aebe29dc30ebfbc058dab/src/libexpr/lexer.l#L87
+    quotesWhenNeeded = x:
+      if x == "<name>" || x == "<name?>" || x == "*"  # special cases needed for documentation
+          || builtins.match ''[a-zA-Z\_][a-zA-Z0-9\_\'\-]*'' x != null
```

Comment:
Maybe add "." here? While there are no stock options that this would affect, option attr paths do sometimes have .s in them, e.g. `services.nginx.virtualHosts."www.example.com"`.

Reply:
oh, of course.